compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12492,6 +12496,100 @@ private STNode parseVarKeyword() {
     }
 
+    /**
+     * Parse list match pattern.
+     * <p>
+     *     <code>
+     *         list-match-pattern := [ list-member-match-patterns ]
+     *         list-member-match-patterns :=
+     *          match-pattern (, match-pattern)* [, rest-match-pattern]
+     *          | [ rest-match-pattern ]
+     *     </code>
+     * </p>
+     *
+     * @return Parsed list match pattern node
+     */
+    private STNode parseListMatchPattern() {
+        startContext(ParserRuleContext.LIST_MATCH_PATTERN);
+        STNode openBracketToken = parseOpenBracket();
+        List<STNode> matchPatternList = new ArrayList<>();
+        STNode restMatchPattern = null;
+        STNode matchPatternMemberRhs;
+
+        do {
+            STToken nextToken = peek();
+            if (nextToken.kind == SyntaxKind.ELLIPSIS_TOKEN) {
+                restMatchPattern = parseRestMatchPattern();
+                break;
+            }
+            STNode matchPatternListMember = parseMatchPattern();
+            matchPatternList.add(matchPatternListMember);
+            matchPatternMemberRhs = parseListMatchPatternMemberRhs();
+
+            if (matchPatternMemberRhs != null) {
+                matchPatternList.add(matchPatternMemberRhs);
+            }
+        } while (matchPatternMemberRhs != null);
+
+        if (restMatchPattern == null) {
+            restMatchPattern = STNodeFactory.createEmptyNode();
+        }
+
+        STNode matchPatternListNode =  STNodeFactory.createNodeList(matchPatternList);
+        STNode closeBracketToken = parseCloseBracket();
+        endContext();
+
+        return STNodeFactory.createListMatchPatternNode(openBracketToken, matchPatternListNode, restMatchPattern,
+                closeBracketToken);
+    }
+
+    /** Parse rest match pattern.
+     *<p>
+     *     <code>
+     *         rest-match-pattern := ... var variable-name
+     *     </code>
+     *</p>
+     *
+     * @return Parsed rest match pattern node
+     */
+    private STNode parseRestMatchPattern() {
+        startContext(ParserRuleContext.REST_MATCH_PATTERN);
+        //We approach here only after seeing ellipsis token hence consume.
+        STNode ellipsisToken = consume();
+        STNode varKeywordToken = parseVarKeyword();
+        STNode variableName = parseVariableName();
+        endContext();
+
+        STSimpleNameReferenceNode simpleNameReferenceNode =
+                (STSimpleNameReferenceNode) STNodeFactory.createSimpleNameReferenceNode(variableName);
+        return STNodeFactory.createRestMatchPatternNode(ellipsisToken, varKeywordToken, simpleNameReferenceNode);
+    }
+
+    private STNode parseListMatchPatternMemberRhs() {
+        return parseListMatchPatternMemberRhs(peek().kind);
+    }
+
+    private STNode parseListMatchPatternMemberRhs(SyntaxKind nextTokenKind) {
+        switch (nextTokenKind) {
+            case COMMA_TOKEN:
+                return parseComma();
+            case CLOSE_BRACKET_TOKEN:
+            case EOF_TOKEN:
```

Comment:
should we add `=>` also here? What if the close bracket is missing?

Reply:
Close bracket will recover there